src/Analyzer.Utilities/Extensions/IMethodSymbolExtensions.cs

```diff
@@ -104,14 +104,41 @@ public static bool IsImplementationOfInterfaceMethod(this IMethodSymbol method,
             INamedTypeSymbol constructedInterface = typeArgument != null ? interfaceType?.Construct(typeArgument) : interfaceType;
             var interfaceMethod = constructedInterface?.GetMembers(interfaceMethodName).Single() as IMethodSymbol;
 
-            if (interfaceMethod != null && method.Equals(method.ContainingType.FindImplementationForInterfaceMember(interfaceMethod)))
+            return IsInterfaceMethodImplementation(method, interfaceMethod);
+        }
+
+        /// <summary>
+        /// Checks if the given method implements any interface method, implicitly or explicitly.
+        /// </summary>
+        public static bool IsImplementationOfAnyInterfaceMethod(this IMethodSymbol method)
```

Comment:
This method just ported over from `CancellationTokenParametersMustComeLast.cs` below